src/scheduler/fifo.c

```diff
@@ -2507,13 +2501,14 @@ parse_sched_obj(struct batch_status *status)
 			long newval;
 
 			newval = res_to_num(attrp->value, NULL);
-			if (sc_attrs.attr_update_period != newval) {
-				sc_attrs.attr_update_period = newval;
+			sc_attrs.attr_update_period = newval;
+			if (newval != prev_attr_u_period) {
 				last_attr_updates = 0;
```

Comment:
i think it might be less confusing if we reset it, e.g - if it's been 4 mins since last update, admin sets the period to 5 mins but we send the first update out after 1 min, it might not be obvious why. Secondary reason: It makes it easier to write automated tests. Let me know if you still feel that we should not do this.

Reply:
I don't agree that it is less confusing.  How I see it is that you are punishing jobs that have been there longer.  If you constantly change this period one second before it triggers, you will never update any jobs.  If you keep it the same, you'll either immediately trigger because you lowered the period, or wait a little longer if you raised it.  You are correct, it is harder to write automated tests.  You can force it to be reset, but it is kind of hacky.  You can set it to 0 and run a cycle, and then set it to what you want.

Reply:
" How I see it is that you are punishing jobs that have been there longer. If you constantly change this period one second before it triggers, you will never update any jobs"

No, we will send out updates when last_attr_updates is 0, so if somebody constantly changes the period one second before it triggers, they will have just triggered it manually. The idea is: if you modify the period, it will be like it's been set for the first time: we will send out the first update, then we will send it N seconds later.

Reply:
Ahh OK, I misunderstood.  I thought we waited the period seconds before sending the updates.  I'm fine with this now.